### PR TITLE
Refactor caliper app code.

### PR DESCRIPTION
**Pull request by danialmalik**
Now CaliperTracking app applies required settings in django if it is included in INSTALLED_APPS list.

**Checks before merge:**

- [ ] Reviewed
- [ ] Commits squashed

### CODE REVIEW

File: openedx/features/caliper_tracking/settings.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+"""
+All settings for Caliper Tracking app.
+"""
+
+LOGIN_EVENT_EMITTER = 'caliper_tracking.helpers.emit_login_event'
+CALIPER_TRACKING_PROCESSOR = 'openedx.features.caliper_tracking.processor.CaliperProcessor'
+
+CALIPER_TRACKING_APP_NAME = 'openedx.features.caliper_tracking'
```
Comment (asamolion) [inline openedx/features/caliper_tracking/settings.py]: Shouldn't the name be only `caliper_tracking`?

File: openedx/features/caliper_tracking/settings.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+"""
+All settings for Caliper Tracking app.
+"""
+
+LOGIN_EVENT_EMITTER = 'caliper_tracking.helpers.emit_login_event'
+CALIPER_TRACKING_PROCESSOR = 'openedx.features.caliper_tracking.processor.CaliperProcessor'
```
Comment (asamolion) [inline openedx/features/caliper_tracking/settings.py]: Please manually test all of this by creating a dummy repo of this and then testing through site packages.

File: openedx/features/caliper_tracking/settings.py (hunk 1 above)
Comment (danialmalik) [reply openedx/features/caliper_tracking/settings.py]: App wasn't loading with only `caliper_tracking`.

Comment (asamolion) [reply openedx/features/caliper_tracking/settings.py]: Test and confirm the installation method from PyPI (python site-packages).

File: openedx/features/caliper_tracking/apps.py [hunk 3]
```diff
@@ -1,41 +1,20 @@
...
-    CALIPER_TRACKING_BACKENDS,
-    CALIPER_TRACKING_PROCESSOR,
-    LOGIN_EVENT_EMITTER
-)
+from .settings import LOGIN_EVENT_EMITTER
 
 
 class CaliperTrackingConfig(AppConfig):
-    name = CALIPER_TRACKING_APP_NAME
+    name = 'openedx.features.caliper_tracking'
```
Comment (asamolion) [inline openedx/features/caliper_tracking/apps.py]: Why is the full path necessary?

Comment (danialmalik) [reply openedx/features/caliper_tracking/apps.py]: right now this path is necessary, but once we transform it into a separate app and move it into site-packages then it won't be necessary

File: openedx/features/ucsd_features/apps.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+class UcsdFeaturesConfig(AppConfig):
+    name = 'openedx.features.ucsd_features'
+    verbose_name = "UCSD Features"
+
+    def ready(self):
+        # To override the settings after third_party_auth.
+        if settings.FEATURES.get('ENABLE_THIRD_PARTY_AUTH'):
+            self._apply_third_party_auth_settings()
+
+    def _apply_third_party_auth_settings(self):
```
Comment (asamolion) [inline openedx/features/ucsd_features/apps.py]: This function is unnecessary.

File: openedx/features/ucsd_features/apps.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+from django.apps import AppConfig
+from django.conf import settings
+
+
+LOGIN_EVENT_EMITTER = 'openedx.features.ucsd_features.utils.emit_login_event'
+
+
+class UcsdFeaturesConfig(AppConfig):
+    name = 'openedx.features.ucsd_features'
```
Comment (asamolion) [inline openedx/features/ucsd_features/apps.py]: Proper naming please.

File: openedx/features/caliper_tracking/settings.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+"""
+All settings for Caliper Tracking app.
+"""
+
+CALIPER_TRACKING_PROCESSOR = 'openedx.features.caliper_tracking.processor.CaliperProcessor'
+
+CALIPER_TRACKING_BACKENDS = {
+    'logger': {
+        'ENGINE': CALIPER_TRACKING_PROCESSOR,
```
Comment (asamolion) [inline openedx/features/caliper_tracking/settings.py]: Why do you need to define a constant if its being referred in the same file?

File: openedx/features/caliper_tracking/settings.py (hunk 2 above)
Comment (asamolion) [reply openedx/features/caliper_tracking/settings.py]: Please use proper path.

File: openedx/features/caliper_tracking/__init__.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
+from django.conf import settings as django_settings
+
+from . import processor
+from .settings import CALIPER_TRACKING_BACKENDS, CALIPER_TRACKING_PROCESSOR
+
+
+default_app_config = 'openedx.features.caliper_tracking.apps.CaliperTrackingConfig'
+
+
+def _enable_caliper_backends():
```
Comment (asamolion) [inline openedx/features/caliper_tracking/__init__.py]: This function also appears to be unnecessary.

File: lms/envs/aws.py [hunk 8]
```diff
@@ -1096,11 +1096,15 @@
...
 ############### Settings for UCSD Support #####################
 SUPPORT_DESK_EMAILS = ENV_TOKENS.get('SUPPORT_DESK_EMAILS')
+INSTALLED_APPS.append('openedx.features.ucsd_features')
 
 ############### Settings for Caliper Tracking #####################
-INSTALLED_APPS += [
-    'openedx.features.caliper_tracking.apps.CaliperTrackingConfig'
-]
+if FEATURES.get('ENABLE_EVENT_CALIPERIZATION'):
+    INSTALLED_APPS.insert(
```
Comment (asamolion) [inline lms/envs/aws.py]: Good.

File: openedx/features/caliper_tracking/settings.py (hunk 6 above)
Comment (danialmalik) [reply openedx/features/caliper_tracking/settings.py]: The same constant is also being used in another file.

File: openedx/features/caliper_tracking/settings.py (hunk 2 above)
Comment (danialmalik) [reply openedx/features/caliper_tracking/settings.py]: we can use proper paths once this is converted into separate app. Then all of this code in `openedx` won't be required. but right now. we have to use these paths.

File: openedx/features/ucsd_features/apps.py (hunk 4 above)
Comment (danialmalik) [reply openedx/features/ucsd_features/apps.py]: refactored.

File: openedx/features/ucsd_features/apps.py (hunk 5 above)
Comment (danialmalik) [reply openedx/features/ucsd_features/apps.py]: This is necessary as `ucsd_features` is in `openedx/features/` and requires the full path to be used.

**Pull request merged**